lib/_http_outgoing.js

```diff
@@ -122,43 +123,47 @@ OutgoingMessage.prototype._send = function(data, encoding, callback) {
 
 
 OutgoingMessage.prototype._writeRaw = function(data, encoding, callback) {
-  if (util.isFunction(encoding)) {
+  if (typeof encoding === 'function') {
     callback = encoding;
     encoding = null;
   }
 
   if (data.length === 0) {
-    if (util.isFunction(callback))
+    if (typeof callback === 'function')
       process.nextTick(callback);
     return true;
   }
 
-  if (this.connection &&
-      this.connection._httpMessage === this &&
-      this.connection.writable &&
-      !this.connection.destroyed) {
+  var connection = this.connection;
```

Comment:
Worst case, it only "saves" one lookup (when connection is falsy); however it's definitely necessary when you get [here](https://github.com/iojs/io.js/pull/605/files#diff-286202fdbdd74ede6f5f5334b6176b5cR149), when you can be looping a lot of times, one char at a time